### PR TITLE
feat(batch): add launchTemplateId in LaunchTemplateSpecification

### LINKED ISSUE #20163: [aws-batch] missing launchTemplateId in LaunchTemplateSpecification
### Describe the feature

Add support for `launchTemplateId` when attaching a launchTemplate to a Batch Compute environment. Currently it requires using a launchTemplateName.

```
new batch.ComputeEnvironment(this, 'ComputeEnv', {
  computeResources: {
    launchTemplate: {
      launchTemplateName: myLaunchTemplate.launchTemplateName as string, //or simply use an existing template name
    },
    vpc,
  }
});
```

This should be supported: https://docs.aws.amazon.com/batch/latest/APIReference/API_LaunchTemplateSpecification.html

### Use Case

I would like to stop naming resources explicitly as per CDK best practice and let CDK handle naming the launch template.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

1.153.1

### Environment details (OS name and version, etc.)

Amazon Linux 2

### PR DESCRIPTION
**Pull request by robertd**
Closes #20163

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by robertd** [comment]
cc @madeline-k  based on gitbot assignment of the https://github.com/aws/aws-cdk/issues/20163 issue

**Review by comcalvi — DISMISSED** [review]
Thanks for the PR!

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts [hunk 1]
```diff
@@ -65,10 +65,18 @@ export enum AllocationStrategy {
  * Launch template property specification
  */
 export interface LaunchTemplateSpecification {
+  /**
+   * The Launch template ID
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-batch/lib/compute-environment.ts:69]:
```suggestion
   * The Launch template ID. Mutually exclusive with `launchTemplateName`.
```

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts [hunk 2]
```diff
@@ -65,10 +65,18 @@ export enum AllocationStrategy {
...
+   *
+   * @default - no launch template id provided
+   */
+  readonly launchTemplateId?: string;
   /**
    * The Launch template name
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-batch/lib/compute-environment.ts:75]:
```suggestion
   * The Launch template name. Mutually exclusive with `launchTemplateId`
```

File: packages/@aws-cdk/aws-batch/README.md [hunk 3]
```diff
@@ -119,7 +119,7 @@ The alternative would be to use the `BEST_FIT_PROGRESSIVE` strategy in order for
 
 Simply define your Launch Template:
 
-```text
+```ts
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-batch/README.md:122]: I don't remember exactly why this example was marked as "typescript only", but if it passes `jsii-rosetta` then I'm fine with changing this now. However, we need to remove the comment that states `// This example is only available in TypeScript`

**Review by kaizencc — COMMENTED** [review]
I have a few nits here... will edit myself

**Review by kaizencc — DISMISSED** [review]
And now approved :)

**Comment by robertd** [comment]
@kaizencc Thanks for stepping in with updates. However, is this PR stuck now in CI/CD?

**Comment by robertd** [comment]
TY @kaizencc

**Pull request merged**